### DISCUSSION TITLE
Microsoft Fabric operators for running spark jobs would be great

### DISCUSSION THREAD

**Discussion by joshdanko** [apache/airflow#38131] [Ideas]
Is something similar to AzureSynapseRunSparkBatchOperator for Fabric in the works?  I am trying to figure out how to run Fabric SPark Jobs from Airflow.  Haven't figured it out yet.

**Comment by Taragolis** [comment]
AFAIK, @ambika-garg working on / investigate about integration with Microsoft Fabric. But I guess there is no ETA when it will be implemented.

**Comment by ambika-garg** [comment]
Hi Andrey,

MS Fabric doesn't support Service principal authentication yet, It will be
released in May, then I will continue working on it. Meanwhile, I am
working on PowerBI Provider.

Regards,
Ambika Garg

**Comment by edgBR** [comment]
Hi,

I think this should be possible now.

The only challenge is probably the base64 encoding for the payload of the job.

https://learn.microsoft.com/en-us/fabric/data-engineering/spark-job-definition-api
